stackslib/src/config/mod.rs

```diff
@@ -1656,52 +1656,231 @@ impl BurnchainConfigFile {
+    /// The publicly accessible URL that this node advertises to peers during the P2P handshake
+    /// as its HTTP RPC endpoint. Other nodes or services might use this URL to query the node's API.
+    ///
+    /// Default: `http://{rpc_bind}` (e.g., "http://0.0.0.0:20443" if rpc_bind is default).
     pub data_url: String,
+    /// The publicly accessible IPv4 address and port that this node advertises to peers for P2P connections.
+    /// This might differ from `p2p_bind` if the node is behind NAT or a proxy.
+    /// Note: The default value derivation might be unexpected, potentially using the `rpc_bind` address; explicit configuration is recommended if needed.
+    ///
+    /// Default: Derived from `rpc_bind` (e.g., "0.0.0.0:20443" if rpc_bind is default).
     pub p2p_address: String,
+    /// The private key seed, provided as a hex string in the config file, used specifically for the
+    /// node's identity and message signing within the P2P networking layer.
+    /// This is separate from the main `node.seed`.
+    ///
+    /// Default: Randomly generated 32 bytes.
     pub local_peer_seed: Vec<u8>,
+    /// A list of initial peer nodes used to bootstrap connections into the Stacks P2P network.
+    /// Peers are specified in a configuration file as comma-separated strings in the
+    /// format `"PUBKEY@IP:PORT"` or `"PUBKEY@HOSTNAME:PORT"`. DNS hostnames are resolved
+    /// during configuration loading.
+    ///
+    /// Default: The effective initial list depends on how the node is started:
+    /// - `mainnet` subcommand: Uses the default Hiro and Stacks mainnet seed nodes
+    /// - `testnet` subcommand: Uses the default Hiro testnet seed node
+    /// - Other subcommands (`start`, `mocknet`, `helium`, etc.): Empty vector `[]`.
     pub bootstrap_node: Vec<Neighbor>,
+    /// A list of peer addresses that this node should explicitly deny connections from.
+    /// Peers are specified as comma-separated strings in the format "IP:PORT" or "HOSTNAME:PORT"
+    /// in the configuration file. DNS hostnames are resolved during configuration loading.
+    ///
+    /// Default: Empty vector `[]`.
     pub deny_nodes: Vec<Neighbor>,
+    /// Flag indicating whether this node should activate its mining logic and attempt to produce Stacks blocks.
+    /// Setting this to `true` typically requires providing necessary private keys (either `node.seed` or `miner.mining_key`).
+    /// It also influences default behavior for settings like `require_affirmed_anchor_blocks`.
+    ///
+    /// Default: `false`
     pub miner: bool,
+    /// Setting this to `true` enables the node to replicate the miner and signer Stacker DBs
+    /// required for signing, and is required if the node is connected to a signer.
+    ///
+    /// Default: `false`
     pub stacker: bool,
+    /// Enables a simulated mining mode, primarily for local testing and development.
+    /// When `true`, *and* [`NodeConfig::miner`] is also `true`, the node may generate blocks
```

Comment:
```suggestion
    /// When `true`, *and* [`node::miner`] is also `true`, the node may generate blocks
```
i'm not sure if we should  document this using the function name vs the toml name of `node`

Reply:
Thanks for explaining. Your plan sounds good to me!

Reply:
In VS Code, I see that these link to the https://docs.rs site, not to the local definition. If that's the case, we should also make an effort to publish a crate so that our docs are available on https://docs.rs.

Reply:
Ah, yes, I see the same in that case. The scenario I tried, where it links to docs.rs is when you mouseover the field somewhere, and the docs show in a popover:
<img width="844" alt="image" src="https://github.com/user-attachments/assets/fedaa46f-38fb-42fd-8efa-efe7955f943c" />

Reply:
This does not need to block this PR. I'll resolve the comment.